### PR TITLE
Docs: Document that combinations may output duplicates.

### PR DESCRIPTION
**Pull request by meltinglava**
Also added one doc test to show the case.

Did this as i needed to test it to check if it did this or not. Might as well put it in the docs to show it.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1123,6 +1123,19 @@ pub trait Itertools : Iterator {
     ///     vec![2, 3, 4],
     ///     ]);
     /// ```
+    ///
+    /// Note: Combinations does not take into accout the equality of the inner values.
```
Comment (jswrenn) [inline src/lib.rs]: Spelling nit: `accout` -> `account`

Wording nit: `the inner values` -> `the iterated values`

File: src/lib.rs [hunk 2]
```diff
@@ -1123,6 +1123,19 @@ pub trait Itertools : Iterator {
...
+    /// ```
+    /// use itertools::Itertools;
+    ///
+    /// let it = vec![1, 2, 2].into_iter().combinations(2);
+    /// itertools::assert_equal(it, vec![
+    ///     vec![1, 2], // Note: thise are the same
```
Comment (jswrenn) [inline src/lib.rs]: Spelling nit: `thise` -> `these`

File: src/lib.rs [hunk 3]
```diff
@@ -1123,6 +1123,19 @@ pub trait Itertools : Iterator {
...
+    ///     vec![1, 2], // Note: thise are the same
```
Comment (jswrenn) [inline src/lib.rs]: Spelling nit: `thise` -> `these`

**Comment by meltinglava** [comment]
There we go. Sorry for taking so long.

**Comment by jswrenn** [comment]
Thank you!

**Pull request merged**